Bump JSOSolvers to 0.14 and in JSOSolvers tutorials

arnavk23 opened the pull request:
(fixes #145)

- Add migration note showing old subsolver_type => new subsolver=:symbol API
- Update solver dictionary examples to use subsolver = :cgls, :crls, :lsqr, :lsmr
- Add explicit TRUNK and TRON examples with subsolver, rtol, atol, max_time
- Add reference to JSOSolvers allowed_subsolvers lists
- Link to Krylov.jl docs for subsolver algorithm details
- Fix grammar and copy edits (heading, wording, plurals)

<!-- Edit this file with the relevant information -->

<!-- Choose one: -->
**New tutorial**
**Updates to tutorial**

- [x] Tutorial name: advanced-jsosolvers
- [x] Confirm that either:
  - There is an issue for this tutorial:
    - #145
  - It was discussed in some other manner.
- [x] Confirm that you have used a tool such as [Grammarly] to verify that the text is correct.
- Wait for the workflow, look at the [preview], and confirm:
  - [ ] The preview has your tutorial listed.
  - [ ] The preview looks like what you want.

[preview]: https://jso.dev/JSOTutorials.jl/
[Grammarly]: https://www.grammarly.com

arnavk23 committed:
docs(advanced-jsosolvers): update TRUNK/TRON subsolver usage to new keyword API (fixes #145)

- Add migration note showing old subsolver_type => new subsolver=:symbol API
- Update solver dictionary examples to use subsolver = :cgls, :crls, :lsqr, :lsmr
- Add explicit TRUNK and TRON examples with subsolver, rtol, atol, max_time
- Add reference to JSOSolvers allowed_subsolvers lists
- Link to Krylov.jl docs for subsolver algorithm details
- Fix grammar and copy edits (heading, wording, plurals)

arnavk23 committed:
Merge branch 'JuliaSmoothOptimizers:main' into fix/advanced-jsosolvers-subsolver-api

arnavk23 committed:
docs(advanced-jsosolvers): add migration note and Krylov.jl docs link for complete #145 fix

- Add migration note at top explaining subsolver_type => subsolver API change
- Link to Krylov.jl documentation for subsolver algorithm details
- Ensures users understand the new API and can find detailed subsolver info

arnavk23 committed:
test: fix runtests.jl to only check tutorial directories

- Filter out non-directory items (like files in scripts/)
- Only check for index.jmd and Project.toml (not generated files)
- Fixes test suite that was failing on both main and fix branches

arnavk23 committed:
Update index.jmd

tmigot commented on test/runtests.jl:
```diff
@@ -5,16 +5,18 @@ function tests()
   )
   for dir in dirs
-    subdirs = readdir(joinpath(rootdir, dir))
+    subdirs = filter(y -> isdir(joinpath(rootdir, dir, y)), readdir(joinpath(rootdir, dir)))
```
Why do any change in this file?

tmigot commented on tutorials/advanced-jsosolvers/index.jmd:
````diff
@@ -38,27 +38,48 @@ For this task, several solvers are available.
+For example, to call TRUNK with an explicit subsolver and tolerances:
+
+```julia
+stats = trunk(nls; subsolver = :cgls)
````
I am very surprised that this work with JSOSolvers 0.11 . Have you tested it?

tmigot commented on tutorials/advanced-jsosolvers/index.jmd:
````diff
@@ -38,27 +38,48 @@ For this task, several solvers are available.
+```julia
+stats = trunk(nls; subsolver = :cgls)
+```
+
````
```suggestion
```

arnavk23 committed:
Update tutorials/advanced-jsosolvers/index.jmd

Co-authored-by: Tangi Migot <tangi.migot@gmail.com>

arnavk23 committed:
Revert to correct JSOSolvers 0.11 API

The previous changes attempted to use `subsolver = :symbol` API which doesn't exist in JSOSolvers 0.11.
Testing confirmed that JSOSolvers 0.11 uses `subsolver_type = Type` (e.g., CglsSolver, LsmrSolver).
This commit reverts to the correct working API that is compatible with JSOSolvers 0.11.2.

arnavk23 committed:
Update runtests.jl

arnavk23 committed:
Update tutorials to use JSOSolvers 0.14 with Symbol-based subsolver API

- Update JSOSolvers from 0.11 to 0.14 in Project.toml files
- Update Krylov from 0.9 to 0.10 (required by JSOSolvers 0.14)
- Replace Type-based subsolver API with Symbol-based API:
  * subsolver_type = CglsSolver → subsolver = :cgls
  * subsolver_type = LsmrSolver → subsolver = :lsmr
  * etc.
- Remove Krylov.jl import (no longer needed for subsolver types)
- Remove compatibility note about JSOSolvers 0.11

This addresses issue #145 by updating to the new subsolver API
introduced in JSOSolvers 0.14.0.

Affected tutorials:
- tutorials/advanced-jsosolvers/
- tutorials/introduction-to-jsosolvers/

Tests: All package tests pass ✓

tmigot commented on tutorials/introduction-to-jsosolvers/Project.toml:
```diff
@@ -12,6 +12,6 @@ SolverCore = "ff4d7338-4cf1-434d-91df-b86cb86fb843"
 CSV = "0.10"
 DataFrames = "1.4"
-JSOSolvers = "0.11"
+JSOSolvers = "0.14"
```
For this tutorial, the problem was the `using JSOSolvers` was define after a cell using it.

arnavk23 replied:
Can you test and send me the error, it is working fine on my end.

tmigot replied:
How? 
<img width="704" height="710" alt="image" src="https://github.com/user-attachments/assets/199ab945-c29b-4acb-a658-9e6594a6096c" />
access JSOSolvers.trunkls_subsolver before `using JSOSolvers`

tmigot reviewed (changes requested):
@arnavk23 have you tested that the scripts work?

arnavk23 committed:
intro tutorial: ensure `using JSOSolvers` precedes constants listing

The version badge on the site is derived from the local Project.toml,
so we test against tutorials/introduction-to-jsosolvers/Project.toml.
This change guarantees the constants are referenced after importing
JSOSolvers (v0.14+ Symbol subsolver API).

tmigot commented on tutorials/introduction-to-jsosolvers/Manifest.toml:
```diff
@@ -0,0 +1,1355 @@
+# This file is machine-generated - editing it directly is not advised
```
Remove this file

tmigot commented on tutorials/introduction-to-jsosolvers/index.jmd:
````diff
@@ -46,6 +46,10 @@ The following examples illustrate this specialization.
 To list the allowed least-squares subsolvers for these specializations:
 
+```julia
+using JSOSolvers
````
Can you add the using JSOSolvers in the next cell?

arnavk23 committed:
Delete tutorials/introduction-to-jsosolvers/Manifest.toml

arnavk23 committed:
Clean up code block formatting in index.jmd

Removed unnecessary code block delimiters from documentation.

arnavk23 committed:
Modify import statements in index.jmd

Updated import statements to include ADNLPModels with JSOSolvers.

tmigot committed:
Update tutorials/advanced-jsosolvers/Project.toml
